loadtest/control/gencontroller/controller.go

```diff
@@ -96,13 +96,37 @@ func (c *GenController) Run() {
 
 	c.status <- control.UserStatus{ControllerId: c.id, User: c.user, Info: "user started", Code: control.USER_STATUS_STARTED}
 
+	cpaEnabled, resp := control.CustomProfileAttributesEnabled(c.user)
+	if resp.Err != nil {
+		c.sendFailStatus("Failed to retreive CustomProfileAttributesEnabled")
+		return
+	}
+
+	// Create CPA fields
+	if cpaEnabled {
+		actions := map[string]userAction{
+			"createCPAField": {
+				run:        c.createCPAField,
+				frequency:  int(c.config.NumCPAFields),
+				idleTimeMs: 1000,
+			},
+		}
+		c.runActions(actions, func() bool {
+			return st.get(StateTargetCPAFields) >= c.config.NumCPAFields
+		})
+	}
+
```

Comment:
I should have been more clear in my comment the other day. This should happen *before* the regular actions but *after* the user has logged in (we can do it right after `initActions`), otherwise they cannot perform any action.